### ISSUE TITLE
What should happen with the focus after someone opened a dropdown (with the mouse)?

### ISSUE THREAD

**Issue by Reinmar** [ckeditor/ckeditor5#2000]
Currently, if you open a dropdown **with your mouse** the focus stays in the editor (of course, it behaves differently when you first focused the toolbar with alt+f10):

![image](https://user-images.githubusercontent.com/156149/63760233-3a10d080-c8bf-11e9-857c-4565522cb807.png)

Which means that we have issues like https://github.com/ckeditor/ckeditor5-font/issues/54.

While the above mentioned problem is indeed an issue and the solution to it seems trivial – let's close the dropdown when the selection changes (CKEditor 4 does that), I'd like to know what should actually happen with focus. Because, I can imagine that it's more useful for the focus to be moved to the dropdown so you can use your keyboard to navigate there (may be useful sometimes).

cc @Comandeer

**Comment by Reinmar** [comment]
cc @oleq @dkonopka How it works in other similar apps?

**Comment by Comandeer** [comment]
Selection in such cases should be moved to the first item in the opened menu, similarly to [ARIA Practices listbox example](https://w3c.github.io/aria-practices/examples/listbox/listbox-collapsible.html).

> How it works in other similar apps?

In Word focus is moved to the opened dropdown.

**Comment by oleq** [comment]
> let's close the dropdown when the selection changes 

👍 to that

> Because, I can imagine that it's more useful for the focus to be moved to the dropdown so you can use your keyboard to navigate there

The focus does move to the toolbar and dropdown items when navigation is initiated **using the keyboard**. Run

```js
setInterval( () => console.log( window.document.activeElement ) , 500 );
```

to see it.

When **using the mouse**, the focus remains when the user clicks buttons/dropdown items. Not sure why we did so but AFAIR it was a deliberate decision. To learn more, some research will be needed.

**Comment by oleq** [comment]
Because we set `tabindex="-1"` in all button and prevent default on mousedown, issues like this one pop up https://github.com/ckeditor/ckeditor5/issues/6656.

I checked how it works. To do that I put the editor in a parent container with `:focus-inside` styles and started observing `editor.ui.focusTracker.isFocused`. 

The most visible difference IMO is the focus style which appears around buttons (and dropdown buttons). Most editors do not display them (which does not make sense IMO) and this is something we would need to address if we went with the new behavior.

Also, when the focus lands in a link URL input CSS does not work (and won't because the link popup is in a different DOM branch). But the focus tracker (logged in the console) works in a more predictable manner. I'm rather interested in the focus tracker than CSS here because it reflects the editor's state more accurately as the UI pieces are scattered across the DOM and CSS will not always work (see: link pop-up case).

## Current behavior

![2020-04-27 10 52 03](https://user-images.githubusercontent.com/1099479/80353309-75f89d00-8875-11ea-80e2-3d976c2bca46.gif)


## After re-enabling toolbar item focus

![2020-04-27 10 51 20](https://user-images.githubusercontent.com/1099479/80353331-7ee96e80-8875-11ea-82de-7b89db2a33bc.gif)


```diff
diff --git a/src/button/buttonview.js b/src/button/buttonview.js
index ebab8bc..7574b8b 100644
--- a/src/button/buttonview.js
+++ b/src/button/buttonview.js
@@ -142,7 +142,7 @@ export default class ButtonView extends View {
                                              bind.if( 'withKeystroke', 'ck-button_with-keystroke' )
                                     ],
                                     type: bind.to( 'type', value => value ? value : 'button' ),
-                                    tabindex: bind.to( 'tabindex' ),
+                                    // tabindex: bind.to( 'tabindex' ),
                                     'aria-labelledby': `ck-editor__aria-label_${ ariaLabelUid }`,
                                     'aria-disabled': bind.if( 'isEnabled', true, value => !value ),
                                     'aria-pressed': bind.to( 'isOn', value => this.isToggleable ? String( value ) : false )
@@ -152,7 +152,7 @@ export default class ButtonView extends View {

                            on: {
                                     mousedown: bind.to( evt => {
-                                             evt.preventDefault();
+                                             // evt.preventDefault();
                                     } ),

                                     click: bind.to( evt => {
```

**Comment by apfelbox** [comment]
Hi,

I am the author of #6656.

> I'm rather interested in the focus tracker than CSS here because it reflects the editor's state more accurately as the UI pieces are scattered across the DOM and CSS will not always work (see: link pop-up case).

That would be my way to go as well. I only used CSS as a band-aid, because I couldn't find anything to listen for in JS at all. So doing this in JS seems way more robust, as it doesn't depend on "correct" nesting of the components (nesting that often would have its own issues).

If I could just easily attach a class to the DOM container if anything inside the toolbar "has focus/is active" and remove it afterwards, that would be superb! 👍

**Comment by akahmis** [comment]
Can someone tell me how to fix this issue? In my case when i focus on editor to type the dropdown appears
